ee/clickhouse/queries/funnels/test/test_funnel.py

```diff
@@ -1463,3 +1463,195 @@ def test_funnel_aggregation_with_groups(self):
         self.assertEqual(result[1]["count"], 1)
         self.assertEqual(len(result[1]["people"]), 1)
         self.assertAlmostEqual(result[1]["average_conversion_time"], 86400)
+
+    @snapshot_clickhouse_queries
+    def test_funnel_group_aggregation_with_groups_entity_filtering(self):
+        GroupTypeMapping.objects.create(team=self.team, group_type="organization", group_type_index=0)
+        GroupTypeMapping.objects.create(team=self.team, group_type="company", group_type_index=1)
+
+        create_group(team_id=self.team.pk, group_type_index=0, group_key="org:5", properties={"industry": "finance"})
+        create_group(team_id=self.team.pk, group_type_index=0, group_key="org:6", properties={"industry": "technology"})
+
+        create_group(team_id=self.team.pk, group_type_index=1, group_key="company:1", properties={})
+        create_group(team_id=self.team.pk, group_type_index=1, group_key="company:2", properties={})
+
+        filters = {
```

Comment:
I gave this feedback earlier as well:

I think the code structure here kind of smells.

We have:

```
setup_groups
filter_object
setup_event_data
run
assert
```

Put the filter data right next to the run instead of having it here? Instead of extracting the hash perhaps have

```
filter = Filter(data=..., team=self.team)
result = ClickhouseFunnel(filter, self.team).run()
```

This applies to other tests as well.

Reply:
whoops, I moved the filter down, but not the data 🤦 . Thanks, will do